tests/testthat/test-merge_sce_list.R

```diff
@@ -417,39 +498,61 @@ test_that("merging SCEs with different altExp features works as expected, with a
     preserve_rowdata_cols = c("gene_names")
   )
 
-  merged_altexp <- altExp(merged_sce)
+  expect_equal(altExpNames(merged_sce), "adt")
 
+})
 
-  expect_true(altExpNames(merged_sce) == altexp_name)
-  expect_equal(dim(merged_altexp), c(num_altexp_features, total_cells))
-  expect_equal(rownames(merged_altexp), full_altexp_features)
 
-  expected_colnames <- sce_list_with_altexp |>
-    purrr::imap(
-      \(sce, sce_name) glue::glue("{sce_name}-{colnames(sce)}")
-    ) |>
-    unlist() |>
-    unname()
-  expect_equal(colnames(merged_altexp), expected_colnames)
+## Other tests ------------------
+
+test_that("build_na_matrix works as expected",{
+
+  rows <- letters[1:5]
+  cols  <- letters[6:10]
+  sparse_mat <- build_na_matrix(
+    "name",
+    rows,
+    cols
+  )
+
+  expect_true(
+    class(sparse_mat) == "dgCMatrix"
+  )
```

Comment:
My bad here: in general it is better to use `is` for these tests, as we want to allow flexibility from parent classes. the `Matrix` constructors can choose among different classes, and we don't really care which specific class uses most of the time.

```suggestion
  expect_true(
    is(sparse_mat, "sparseMatrix")
  )
```

But even better: 
```suggestion
  expect_s4_class(sparse_mat, "sparseMatrix")
```

Reply:
I seem to have landed on `expect_s4_class(sparse_mat, "dgCMatrix")`?

```
expect_s4_class(sparse_mat, "sparseMatrix") is not TRUE

`actual` is a double vector (NA, NA, NA, NA, NA, ...)
`expected` is a logical vector (TRUE)
```

Reply:
Sticking with ` is(sparse_mat, "dgCMatrix")` for now.

Reply:
Note my second suggestion _replaces_ the `expect_true()` and does not nest in it.

Reply:
https://github.com/AlexsLemonade/scpcaTools/pull/251/commits/4f61572b84623b6ed19f7108fc44ec447360213f ✅